src/gui/sdlmain.cpp

```diff
@@ -1644,27 +1824,31 @@ static void GUI_StartUp(Section * sec) {
 
 	std::string output=section->Get_string("output");
 
-	if (output == "surface") {
-		sdl.desktop.want_type=SCREEN_SURFACE;
-	} else if (output == "texture") {
-		sdl.desktop.want_type=SCREEN_TEXTURE;
-		SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, "linear");
-	} else if (output == "texturenb") {
-		sdl.desktop.want_type=SCREEN_TEXTURE;
-		// Currently the default, but... oh well
-		SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, "nearest");
-#if C_OPENGL
-	} else if (output == "opengl") {
-		sdl.desktop.want_type=SCREEN_OPENGL;
-		sdl.opengl.bilinear=true;
-	} else if (output == "openglnb") {
-		sdl.desktop.want_type=SCREEN_OPENGL;
-		sdl.opengl.bilinear=false;
-#endif
-	} else {
+	if (!ParseOutType(
+		output.c_str(),
+		&sdl.desktop.want_type,
+		&sdl.kind)
+	) {
```

Comment:
Your changed code does not respect C_OPENGL ifdef.

It would be so much easier (to write, read, and to review) to simply add:
```
else if (output == "texturepp") {
    sdl.desktop.want_type = SCREEN_TEXTURE;
    sdl.kind = OkPerfect;
}
```

instead of adding this whole parsing code (which we will remove during the transition to separate setting anyway).

Reply:
> Your changed code does not respect C_OPENGL ifdef.

It is not a problem because the availability of *OpenGL* is taken care of in the list of valid values:

~~~
   const char *outputs[] = {
      "surface",
      "texture",
      "texturenb",
      "texturepp",
#if C_OPENGL
      "opengl",
      "openglnb",
#endif
      0
   };
~~~

I see no reason why the code should repeat itself and repeat the same checks in multiple places. The parser function can be general, whereas the business logic will do the validation.

Reply:
Hmm. Until now I followed the upstream closely in placing all, even slightly, OpenGL related code inside ifdef, so it would be possible to build DOSBox without any OpenGL support… I don't really like C_OPENGL ifdef and would prefer to not to have it at all… but it does improve portability for platforms, that don't support OpenGL:

- Mobile and embedded devices, that support OpenGL ES only
- Future macOS, when Apple will inevitably drop OpenGL support
- WebGL-only platforms (or other future Web graphics APIs)

Overall in other comments I requested reverting code to the original litany of if-elses, so that should take care of this request as well.